### ISSUE TITLE
not sure when the sms should arrive

### ISSUE THREAD

**Issue by johba37** [travisdmathis/leap-watcher#1]
had a look through the code. the submission looks great. I wanted to make sure that I get everything tested at lest once, and hit a little bump:

1. cloned and installed (also installed reddis and started server)
2. got a twillio account and bought a number
3. configured all that and testnet + rinkeby in the config

startup looks great. then went to page and registered a number.

in the logs i see:

```
[1556829756048] INFO  (34025 on a1708.fritz.box): ##### FINISHED PLASMA CHECK #####
[1556829816055] INFO  (34025 on a1708.fritz.box): ##### STARTING PLASMA CHECK #####
[1556829816055] INFO  (34025 on a1708.fritz.box): 
[1556829816412] INFO  (34025 on a1708.fritz.box): UNSPENT TRANSACTIONS: 1435
[1556829816530] INFO  (34025 on a1708.fritz.box): ADDRESSES: 2
[1556829816650] INFO  (34025 on a1708.fritz.box): COLORS: 2
[1556829816653] INFO  (34025 on a1708.fritz.box): PLASMA UTXO SUM: 7564.90221
[1556829818561] INFO  (34025 on a1708.fritz.box): PLASMA EXITS SUM: 0
[1556829819592] INFO  (34025 on a1708.fritz.box): PLASMA BALANCE: 7585.35009
[1556829819592] ERROR (34025 on a1708.fritz.box): VALID PLASMA: false
```

so i expect to receive a number. but nothing came. is this a problem with my twillio config?

**Comment by johba37** [comment]
i think i get it:
https://github.com/travisdmathis/leap-watcher/blob/fed2f9d4baf526bc431644e3ab6bed77d20fa941/server/lib/watcher.js#L151-L153

you only send the sms 1 time when the state transitions from `true` to `false`.
what is a good way to verify that? 🤔

**Comment by travisdmathis** [comment]
@johannbarbie yes, for now it sends 1 time and then stops until the error has been cleared.. I didn't wanna spam every check and burn up my free sms or spam subscribers.. you'd need to further implement some checks if you wanted to send more than 1 SMS message.  Otherwise you could incur some massive charges if you had a lot of subscribers and possibly make them angry w/ constant SMS spams.

How to retest:
**Option 1:** you make the plasma chain go valid by adding the difference in so UTXO and the Balance are the same it should clear and then when it fails again, it will send another SMS.

https://github.com/travisdmathis/leap-watcher/blob/fed2f9d4baf526bc431644e3ab6bed77d20fa941/server/lib/watcher.js#L147

**Option 2:** You can also flush your redis db and rerun it to get the SMS to send again. 
`redis-cli flushall`


It could also be a problem with twilio.. you need to make sure you added the twilio number you purchased to the app config.. as well as whitelist your phone number if you're still using the free trial account.
